Userland/Libraries/LibWeb/Layout/ImageBox.cpp

```diff
@@ -22,6 +22,13 @@ ImageBox::ImageBox(DOM::Document& document, DOM::Element& element, NonnullRefPtr
 
 ImageBox::~ImageBox() = default;
 
+void ImageBox::visit_edges(JS::Cell::Visitor& visitor)
+{
+    Base::visit_edges(visitor);
+    auto const* html_element = dynamic_cast<HTML::HTMLElement const*>(&m_image_provider);
```

Comment:
Instead of using `dynamic_cast` which is very very slow, we can add a virtual to `ImageProvider`. Something like `virtual NonnullGCPtr<HTMLElement> to_html_element() const` and just `return *this;` in all the implementations.